### ISSUE TITLE
[BUG] npm install keeps spawning pwsh processes, consumes all memory, and seem to never finish

### ISSUE THREAD

**Issue by andradei** [npm/cli#6709]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

Running `npm install` via another tool that leverages it ((wails)[https://wails.io/docs/introduction]) on Windows 10 and 11 causes it to use all memory and keeps spawning `pwsh` subprocesses indefinitely.

![image](https://github.com/npm/cli/assets/2653546/04407f41-a236-4e91-b4ac-e8676f5df51a)

### Expected Behavior

`npm install` would install the dependencies and exit successfully.

### Steps To Reproduce

1. Windows 11 or 10 Command Prompt
2. Install [Go](https://go.dev/dl/), than [Wails](https://wails.io/docs/gettingstarted/installation)
3. Create a wails project: `wails init -n wails-test`
4. `cd` into it and run `wails dev`

### Environment

- npm: 9.6.7
- Node.js: v18.17.1
- OS Name: Windows 11 (tried on 10 as well)
- System Model Name: 22H2
- npm config:
```ini
; "builtin" config from C:\Users\me\scoop\apps\nodejs-lts\18.17.1\node_modules\npm\npmrc

cache = "C:\\Users\\me\\scoop\\persist\\nodejs-lts\\cache"
prefix = "C:\\Users\\me\\scoop\\persist\\nodejs-lts\\bin"

; node bin location = C:\Users\me\scoop\apps\nodejs-lts\current\node.exe
; node version = v18.17.1
; npm local prefix = C:\Users\me
; npm version = 9.6.7
; cwd = C:\Users\me
; HOME = C:\Users\me
; Run `npm config ls -l` to show all defaults.
```

**Comment by kchindam-infy** [comment]
@andradei I attempted to repro the issue, but i was unable to observe the described behavior. 
Environment: windows 11, node 18.20, npm 10.7.1 , wails 2.9.1. 
I followed the steps you provided:
Installed GO and wails
Created new wails prject  with wails init -n wails-test
ran wails dev in the project directory

Environment started successfully and i was able to interact with the UI. Did not see excessive memory usage or multiple pwsh processes being spawned and the installation process completed without apparent issues.
